statsmodels/tsa/arima/estimators/hannan_rissanen.py

```diff
@@ -0,0 +1,196 @@
+        Moving average order
+    demean : bool, optional
+        Whether to estimate and remove the mean from the process prior to
+        fitting the ARMA coefficients. Default is True.
+    initial_ar_order : int, optional
+        Order of long autoregressive process used for initial computation of
+        residuals.
+    unbiased: bool, optional
+        Whether or not to apply the bias correction step. Default is True.
+
+    Returns
+    -------
+    parameters : SARIMAXParams object
+    other_results : Bunch
+        Includes three components: `spec`, containing the
+        `SARIMAXSpecification` instance corresponding to the input arguments;
+        `initial_ar_order`, containing the autoregressive lag order used in the
+        first step; and `resid`, which contains the computed residuals from the
+        last step.
+
+    Notes
+    -----
+    The primary reference is [1]_, section 5.1.4, which describes a three-step
+    procedure that we implement here.
+
+    1. Fit a large-order AR model via Yule-Walker to estimate residuals
+    2. Compute AR and MA estimates via least squares
+    3. (Unless `unbiased=False`) Perform bias correction
+
+    The order used for the AR model in the first step may be given as an
+    argument. If it is not, we compute it as suggested by [2]_.
+
+    The estimate of the variance that we use is computed from the residuals
+    of the least-squares regression and not from the innovations algorithm.
+    This is because our fast implementation of the innovations algorithm is
+    only valid for stationary processes, and the Hannan-Rissanen procedure may
+    produce estimates that imply non-stationary processes. To avoid
+    inconsistency, we never compute this latter variance here, even if it is
+    possible. See test_hannan_rissanen::test_brockwell_davis_example_517 for
+    an example of how to compute this variance manually.
+
+    This procedure assumes that the series is stationary.
+
+    TODO: better warnings / errors when series are short relative to lag length
+
+    TODO: add support for fixed parameters
+
+    TODO: seems like we could add support for seasonal parameters here?
+
+    References
+    ----------
+    .. [1] Brockwell, Peter J., and Richard A. Davis. 2016.
+       Introduction to Time Series and Forecasting. Springer.
+    .. [2] Gomez, Victor, and Agustin Maravall. 2001.
+       "Automatic Modeling Methods for Univariate Series."
+       A Course in Time Series Analysis, 171–201.
+
+    """
+    spec = SARIMAXSpecification(endog, ar_order=ar_order, ma_order=ma_order)
+    endog = spec.endog
+    if demean:
+        endog = endog - endog.mean()
+
+    p = SARIMAXParams(spec=spec)
+
+    nobs = len(endog)
+    max_ar_order = spec.max_ar_order
+    max_ma_order = spec.max_ma_order
+
+    # Default initial_ar_order is as suggested by Gomez and Maravall (2001)
+    if initial_ar_order is None:
+        initial_ar_order = max(np.floor(np.log(nobs)**2).astype(int).item(),
```

Comment:
why `.item()`

Reply:
Hmm good question. I wanted an `int` at some point, but can't remember why. Tests pass without it, so I'll remove.